src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,133 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final Object[] DEFAULT_CAPACITY_EMPTY_ELEMENT_DATA = {};
+    private static final int DEFAULT_CAPACITY = 10;
+    private Object[] elementData;
+    private int size = 0;
+
+    public ArrayList() {
+        this.elementData = DEFAULT_CAPACITY_EMPTY_ELEMENT_DATA;
```

Comment:
what about this ? 
Are you sure we need the external array?
```suggestion
    public ArrayList() {
        this.elementData = new Object[DEFAULT_CAPACITY];
```